### PR TITLE
Legend: Better support for 2 y-axis

### LINKED ISSUE #66704: TimeSeries panel with 2 y-axes: Can't tell which y-axis I should read from
## What happened:

In the Billing/Usage dashboard of GrafanaCloud we have various Usage panels which have 2 y-axes:

eg. **Total Billable Metrics Series** and **Total Billable Logs Usage**

image

But how can a user tell whether a line on the chart should be read via the left or the right axis? What units is it in?

Well with the old Graph panel, when the legend is rendered below, we align the items in the legend to the left or to the right to show which axis they should be read with.

But on the new TimeSeries panel, all legend items are aligned left, so I can't tell which lines are in Dollars and which are in the other Y-Axes (one of Million Series, or TiB, etc).

image

For me this is a deal-breaker. We need to show at first glance (and without requiring any further user-interaction such as clicking legend items, which they might not discover by themselves) which lines should be read from which axis.

I don't think we need to reimplement the exact same solution as we did for the Graph panel - honestly that solution is clunky and ugly. We can look to other Data Viz products for other ideas. Googling for images of Google Sheets produced this example:

image

## What you expected to happen:

I should be able to see clearly from the Viz which lines are in dollars and which are not

## How to reproduce it (as minimally and precisely as possible):

Unfortunately I haven't got exact repro steps yet. I tried to repro with the TestDataDB datasource, but created this:

image

So there's something about [this panel on Ops](https://ops.grafana-ops.net/d/2OQseaNMz/billing-usage-current?orgId=1&viewPanel=7) which means the legend items all end up on the left, and none on the right.

## Anything else we need to know?:

Relates to:
- https://github.com/grafana/grafana/issues/49670
-

### PR DESCRIPTION
**Pull request by adela-almasan**
Added a 15px gap between left and right sections; right section moves below left at 50% width. 

https://github.com/grafana/grafana/assets/88068998/bdffed7d-efd6-4a27-b467-79e1aaa4d4c2

Fixes #66704 

**Special notes for your reviewer:**
* More improvements will probably come with the Legends project.
* [debug-Multiple Y-Axes (more than 2!)-2023-10-16 14_45_00.json.txt](https://github.com/grafana/grafana/files/12921697/debug-Multiple.Y-Axes.more.than.2.-2023-10-16.14_45_00.json.txt)

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by adela-almasan** [comment]
cc @samjewell, @lukasztyrala

**Comment by samjewell** [comment]
Thanks for this Adela 👏 . SO glad to see us start to make progress on this.

image
For me, the alignment is still wrong at this screen-width in your demo. The legend-items at the bottom should all be right-aligned, not left-aligned at this point.

Having said that, I LOVE how this works on wider screens as you've demoed it. I think it's an improvement (for wide screens only) on what existed before, in the (Old) Graph panel.

**Question:**

Why change the behaviour at all, at 50% width? Couldn't you instead do something like:

- Split the available legend-space below the graph into 50/50 left and right. This would naturally update for all screen sizes
- Then render the left-axis legend items into the left-legend, and the right-axis items into the right-legend?

I dunno - maybe that would highlight some additional needs - for vertical scrolling inside the legend for example, or for ellipsing long legend-texts to be revealed on hover, or other similar changes. But maybe these are things we should be doing anyway.

**Comment by leeoniya** [comment]
@samjewell i hope that chart is theoretical 😅. we cannot save people from making bonkers things. 100 axes with 1000 series isnt something we're gonna spend too much time worrying about. i think 4 axes total with maybe 3 max on one side and maybe 30ish series feels like a reasonable goal.

> Why change the behaviour at all, at 50% width? Couldn't you instead do something like:

maybe cause if all you have is a left axis or you have 30 series on a left axis and just one on the right you would not be using 50% of the available space and would be forced into scrolling or overflow too prematurely?

**Comment by samjewell** [comment]
@leeoniya thanks very much for your last comments. At first I was taken aback, and I suppose preparing myself to argue. But actually I think you've made great points.

Sorry I didn't reply sooner - it's been hectic with both kids getting sick, and some production incidents this month.

> @samjewell i hope that chart is theoretical 😅. we cannot save people from making bonkers things 

Actually I just took a single screenshot from the video that Adela posted in the PR description! Her chart, not mine. No idea if it's theoretical (but I assume so). That said, your comment made me realise that possibly I've made some bonkers things too, or have done in conjunction with every other committer on the Billing dashboard! I'll see if I can make them less bonkers 🤞 

> maybe 30ish series feels like a reasonable goal.

Thanks. I'll consider this too in that case. I completely see that we have to draw the line somewhere!

> you have 30 series on a left axis and just one on the right you would not be using 50% of the available space and would be forced into scrolling or overflow prematurely?

Not sure I'm convinced yet, but that sounds reasonable :-)

File: packages/grafana-ui/src/components/VizLegend/VizLegendList.tsx [hunk 1]
```diff
@@ -109,13 +109,15 @@ const getStyles = (theme: GrafanaTheme2) => {
       justifyContent: 'space-between',
       width: '100%',
       paddingLeft: theme.spacing(0.5),
+      gap: 15,
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/VizLegend/VizLegendList.tsx:112]: maybe

```suggestion
      gap: '15px 25px',
```

**Review by leeoniya — APPROVED** [review]
lgtm :shipit:

**Pull request merged**